packages/block-editor/src/components/link-control/search-item.js

```diff
@@ -35,6 +36,9 @@ function SearchItemIcon( { isURL, suggestion } ) {
 		if ( suggestion.type === 'page' && suggestion.isFrontPage ) {
 			icon = home;
 		}
+		if ( suggestion.type === 'page' && suggestion.isBlogHome ) {
```

Comment:
I guess we can use an else if to avoid checking the second and a main if that checks for the type page.